src/App.tsx

```diff
@@ -1,60 +1,133 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+  const [titleErrorMessage, setHasTitleErrorMessage] = useState('');
+  const hasTitleError = titleErrorMessage !== '';
+
+  function getNewPostId(todosList: Todo[]) {
+    const maxId = Math.max(...todosList.map(todo => todo.id));
+
+    return maxId + 1;
+  }
+
+  const isValidTitle = (inputedValue: string) => {
+    const regex = /^[a-zA-Zа-яА-ЯїЇєЄіІґҐ\s]*$/;
+
+    return regex.test(inputedValue);
+  };
+
+  const addNewTodo = (newTodo: Todo) => {
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    setTodos(todos => [...todos, newTodo]);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (evnt: React.FormEvent) => {
+    evnt.preventDefault();
+
+    let titleError = '';
+    let userError = false;
+
+    if (!title.trim()) {
+      titleError = 'Please enter a title';
+    } else if (!isValidTitle(title)) {
+      titleError = 'Input should contain only UA and EN';
+    }
+
+    if (userId === 0) {
+      userError = true;
+    }
+
+    setHasTitleErrorMessage(titleError);
+    setHasUserError(userError);
+
+    if (titleError || userError) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getNewPostId(todos),
+      title,
+      userId,
+      completed: false,
+    };
+
+    addNewTodo(newTodo);
+    reset();
+  };
+
+  const handleUserChange = (evnt: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+evnt.target.value);
+
+    setHasUserError(+evnt.target.value === 0);
+  };
+
+  const handleTitleChange = (evnt: React.ChangeEvent<HTMLInputElement>) => {
+    const input = evnt.target.value;
+
+    setTitle(input);
+    if (input.trim()) {
+      setHasTitleErrorMessage('');
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
```

Comment:
[CODE KNOWLEDGE] - The `action` and `method` attributes are not necessary when you handle the form submission via JavaScript. You can remove them to clean up your code as per checklist item 6.